### ISSUE TITLE
DASK to lower memory for merges

### ISSUE THREAD

**Issue by oerbilgin** [biorack/magi#27]
During the actual MAGI integration, the memory stays pretty low, except for the multiprocessing step (short spike, just because of the way python mp works), but then during the merging, the memory goes crazy (table becomes huge), and then another spike during the final score calculation
![screen shot 2017-10-02 at 10 02 51 am](https://user-images.githubusercontent.com/16328959/31089481-276a2398-a759-11e7-9d45-96b7473a012e.jpg)

Use DASK to lower memory usage during merge

**Comment by benbowen** [comment]
@oerbilgin when did you go to Rome?  I worked on it while you were gone, but can't find the code.

**Comment by oerbilgin** [comment]
Last 2 weeks of June

**Comment by benbowen** [comment]
I added an example to the [repo here](https://github.com/biorack/magi/blob/master/prototyping/use_dask_example.ipynb). Briefly, merge works like this:

```
import dask.dataframe as dd
left1 = dd.from_pandas(compound_to_reaction, npartitions=8)
right1 = dd.from_pandas(reaction_to_gene_top, npartitions=4)
compound_to_gene = dd.merge(left1, right1, on='reaction_id', how='left').compute()
```

In addition to the merge commands in this notebook, I tried a few methods to save the dataframe and all created tables that were too big to deal with.  Those should be ignored for now.  Just an FYI, as we go to much larger tables, they will export to files that nobody will be able to open without some custom code.
